connect/models/marketplace.py

```diff
@@ -57,49 +76,71 @@ class AgreementSchema(BaseSchema):
     description = fields.Str()
     created = fields.DateTime()
     updated = fields.DateTime()
-    owner = fields.Nested(CompanySchema, only=('id', 'name'))
-    stats = fields.Dict()
-    active = fields.Bool()
+    owner = fields.Nested(CompanySchema)
+    stats = fields.Nested(AgreementStatsSchema, allow_none=True)
+    author = fields.Nested(CompanySchema, allow_none=True)
```

Comment:
Company? Should be user or contact

Reply:
Ok, I used Company because it has the same fields as owner, if and name. I felt it was not necessary to create an additional User model with the exact same fields. Contact won't do, since it has first_name and last_name, but no name.

Should I create a User model and schema then? Please confirm which of the following fields should be User and which ones should be Company:

- Connection.provider
- Connection.vendor
- EventInfo.by (now that I see this, should EventInfo be renamed Event?)
- Hub.company (seems obvious that it should be Company in this case, but just making sure)
- Marketplace.owner
- Agreement.owner
- Agreement.author (you said it's User in this case)
- Contract.owner
- Contract.creator
- Contract.signee
- TierConfigRequest.assignee
- usage.File.vendor
- usage.File.provider
- usage.Listing.vendor
- usage.Listing.provider

Reply:
.provider/.vendor/.owner/.company - Company
.author/.creator/.signee/.<something>.by - User
https://confluence.int.zone/display/CONNECT/User+Object

Reply:
Done.